webroot/src/components/Users/UserList/UserList.ts

```diff
@@ -0,0 +1,284 @@
+            await this.$store.dispatch('sorting/updateSortOption', {
+                sortingId: listId,
+                newOption: defaultSortOption.value,
+            });
+        }
+
+        if (!direction) {
+            await this.$store.dispatch('sorting/updateSortDirection', {
+                sortingId: listId,
+                newDirection: SortDirection.asc,
+            });
+        }
+    }
+
+    async setDefaultPaging(shouldForce = false): Promise<void> {
+        const { listId } = this;
+        const { page, size } = this.paginationStore.paginationMap[this.listId] || {};
+        const { prevLastKey } = this.usersStore;
+
+        if (!page || shouldForce) {
+            await this.$store.dispatch('pagination/updatePaginationPage', {
+                paginationId: listId,
+                newPage: DEFAULT_PAGE,
+            });
+        }
+
+        if (!size) {
+            await this.$store.dispatch('pagination/updatePaginationSize', {
+                paginationId: listId,
+                newSize: DEFAULT_PAGE_SIZE,
+            });
+        }
+
+        if (prevLastKey) {
+            this.prevKey = prevLastKey;
+        }
+    }
+
+    async fetchListData(search = ''): Promise<void> {
+        const sorting = this.sortingStore.sortingMap[this.listId];
+        const { option, direction } = sorting || {};
+        const pagination = this.paginationStore.paginationMap[this.listId];
+        const { page, size } = pagination || {};
+        const requestConfig: any = {};
+
+        if (option) {
+            const serverSortByMap = {
+                firstName: 'givenName',
+                lastName: 'familyName',
+                permissions: 'permissions',
+                affiliation: 'affiliation',
+                states: 'jurisdiction',
+                accountStatus: 'status',
+            };
+
+            requestConfig.sortBy = serverSortByMap[option];
+        }
+
+        if (direction) {
+            const serverSortDirectionMap = {
+                asc: 'ascending',
+                desc: 'descending',
+            };
+
+            requestConfig.sortDirection = serverSortDirectionMap[direction];
+        }
+
+        // Handle prev / next pages for server paging keys
+        if (page && !this.usersStore.error) {
+            if (this.nextKey && page !== 1) {
+                requestConfig.getNextPage = true;
+            } else if (this.prevKey) {
+                requestConfig.getPrevPage = true;
+            }
+        }
+
+        requestConfig.compact = this.userStore.currentCompact?.type;
+
+        if (search) {
+            requestConfig.search = search;
+        }
+
+        await this.$store.dispatch('users/getUsersRequest', {
+            params: {
+                ...requestConfig,
+                pageNum: page,
+                pageSize: size,
+            }
+        });
+
+        // If we've reached the end of paging
+        if (this.usersStore.error instanceof PageExhaustError && page > 1) {
```

Comment:
Why do you need both boolean clauses?

Reply:
Just to narrow in on the case where the user tried to advance, but we encountered an exhaust. I updated the comment on the line above to clarify it's the user-initiated action states(s) we're looking for.

Reply:
Got it!